src/main/java/net/minecraftforge/common/extensions/IForgeLevel.java

```diff
@@ -34,4 +46,42 @@ public default Collection<PartEntity<?>> getPartEntities()
     {
         return Collections.emptyList();
     }
+
+    /**
+     * Retrieve a block capability.
+     */
+    @Nullable
+    default <T, C> T getCapability(BlockCapability<T, C> cap, BlockPos pos, C context)
+    {
+        return cap.getCapability(self(), pos, null, null, context);
+    }
+
+    /**
+     * Retrieve a block capability. Use this override if you already know the block state and/or the block entity.
+     */
+    @Nullable
+    default <T, C> T getCapability(BlockCapability<T, C> cap, BlockPos pos, @Nullable BlockState state, @Nullable BlockEntity blockEntity, C context)
+    {
+        return cap.getCapability(self(), pos, state, blockEntity, context);
+    }
+
+    /**
+     * Notify all listeners that the capabilities at a specific position might have changed.
+     * This includes new capabilities becoming available.
+     * <p>
+     * This will only do something on {@link ServerLevel}s.
+     * <p>
+     * If you already have a block entity at that position, you can call {@link BlockEntity#invalidateCapabilities()} instead.
+     */
+    default void invalidateCapabilities(BlockPos pos) {
+    }
+
+    /**
+     * Notify all listeners that the capabilities at all the positions in a chunk might have changed.
+     * This includes new capabilities becoming available.
+     * <p>
+     * This will only do something on {@link ServerLevel}s.
+     */
+    default void invalidateCapabilities(ChunkPos pos) {
+    }
```

Comment:
This is a style difference; but why the choice to implement these at the `Level` interface anyway? Would it not always be an incorrect usage to call `invalidateCapabilities()` on a `ClientLevel` or `Level` without checking that it is the server? As opposed to implementing these only on `ServerLevel`:

```java
// typical usage becomes more explicit
if (level instanceof ServerLevel serverLevel) serverLevel.invalidateCapabilities();

// Ambiguous usage is not legal
level.invalidateCapabilities(); // doesn't do anything if it's client

// Calling on a explicit client level that would've compiled before, but been semantically wrong, is prevented
clientLevel.invalidateCapabilities(); // error
```

Reply:
I think I moved it from ServerLevel to Level when I had to implement invalidation for composters. The side checks there looked a bit pointless. There's a lot of precedent for methods that don't do anything on one of the level classes. Mostly a stylistic choice I agree.